File: MachineProblem/Hard/Hard1.cpp
#include <bits/stdc++.h>
/*
    HARD 1 – Student Grade Analyzer
    Goal: Read number of students and their grades; print average, highest, lowest.
    Input:
        n
        g1 g2 ... gn
    Output:
        Average: <avg>\nHighest: <high>\nLowest: <low>
    Example:
        Input: 5\n87 90 78 92 88\n
*/
using namespace std;
int main(){
    int n; if(!(cin>>n)) return 0;
    vector<int> g(n);
    for(int i=0;i<n;i++) cin>>g[i];
    long long sum = 0;
    int high = INT_MIN, low = INT_MAX;
    for(int x: g){ sum += x; if(x>high) high = x; if(x<low) low = x; }
    cout << "Average: " << (sum / n) << endl;
    cout << "Highest: " << high << endl;
    cout << "Lowest: " << low << endl;
    return 0;
}

File: MachineProblem/Hard/Hard1.java
import java.util.*;
public class Main {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        if(!sc.hasNextInt()) return;
        int n = sc.nextInt();
        int[] g = new int[n];
        long sum = 0;
        int high = Integer.MIN_VALUE, low = Integer.MAX_VALUE;
        for(int i=0;i<n;i++){
            g[i] = sc.nextInt();
            sum += g[i];
            if(g[i] > high) high = g[i];
            if(g[i] < low) low = g[i];
        }
        System.out.println("Average: " + (sum / n));
        System.out.println("Highest: " + high);
        System.out.println("Lowest: " + low);
    }
}
